pkg/liqoctl/move/movevolume_test.go

```diff
@@ -0,0 +1,614 @@
+		}), Entry("should return an error if a pod is mounting the volume", mounterTestcase{
+			client: fake.NewClientBuilder().WithObjects(
+				newPod("pod1", "default", []string{"pvc1"}),
+				newPod("pod2", "default", []string{}),
+				newPod("pod3", "default", []string{"pvc3"}),
+				newPod("pod3", "ns2", []string{"pvc3"}),
+				newPod("pod4", "default", []string{"pvc4"}),
+			).Build(),
+			pvc: newPvc("pvc3"),
+			expectedErr: MatchError(fmt.Errorf("the volume (%s/%s) must not to be mounted by any pod, but found mounter pod %s/%s",
+				"default", "pvc3", "default", "pod3")),
+		}))
+
+		type isLocalVolumeTestcase struct {
+			client        client.Client
+			pvc           *corev1.PersistentVolumeClaim
+			expectedErr   OmegaMatcher
+			expectedLocal OmegaMatcher
+			expectedNode  OmegaMatcher
+		}
+
+		DescribeTable("isLocalVolume function", func(c isLocalVolumeTestcase) {
+			local, node, err := isLocalVolume(ctx, c.client, c.pvc)
+			Expect(err).To(c.expectedErr)
+			Expect(local).To(c.expectedLocal)
+			Expect(node).To(c.expectedNode)
+		}, Entry("should return error if no node annotation is found", isLocalVolumeTestcase{
+			client:        fake.NewClientBuilder().Build(),
+			pvc:           newPvc("pvc1"),
+			expectedErr:   HaveOccurred(),
+			expectedLocal: BeFalse(),
+			expectedNode:  BeNil(),
+		}), Entry("should return true if the node is local", isLocalVolumeTestcase{
+			client:        fake.NewClientBuilder().WithObjects(newNode("node1", true)).Build(),
+			pvc:           addNodeMount(newPvc("pvc1"), "node1"),
+			expectedErr:   BeNil(),
+			expectedLocal: BeTrue(),
+			expectedNode:  &matchObject{name: "node1"},
+		}), Entry("should return false if the node is not local", isLocalVolumeTestcase{
+			client:        fake.NewClientBuilder().WithObjects(newNode("node1", false)).Build(),
+			pvc:           addNodeMount(newPvc("pvc1"), "node1"),
+			expectedErr:   BeNil(),
+			expectedLocal: BeFalse(),
+			expectedNode:  &matchObject{name: "node1"},
+		}))
+
+		Context("recreatePvc function", func() {
+
+			var (
+				cl     client.Client
+				oldPvc *corev1.PersistentVolumeClaim
+			)
+
+			BeforeEach(func() {
+				oldPvc = newPvc("pvc1")
+				oldPvc.Spec.VolumeName = "pv1"
+				cl = fake.NewClientBuilder().WithObjects(oldPvc).Build()
+			})
+
+			It("should recreate the PVC with an empty volume name", func() {
+				pvc, err := recreatePvc(ctx, cl, oldPvc)
+				Expect(err).To(BeNil())
+				Expect(pvc.Name).To(Equal(oldPvc.Name))
+				Expect(pvc.Namespace).To(Equal(oldPvc.Namespace))
+				Expect(pvc.Spec.VolumeName).To(BeEmpty())
+			})
+		})
+	})
+
+	Context("forge jobs", func() {
+
+		Context("createSnapshotterJob", func() {
+
+			var (
+				cl  client.Client
+				pvc *corev1.PersistentVolumeClaim
+			)
+
+			BeforeEach(func() {
+				pv := newPv("pv1")
+
+				pvc = newPvc("pvc1")
+				pvc.Spec.VolumeName = pv.Name
+
+				cl = fake.NewClientBuilder().WithObjects(pv, pvc).Build()
+			})
+
+			It("creates a snapshotter job", func() {
```

Comment:
Would suggest to split this It clause in multiple ones, to simplify pin-pointing a problem in case one of the checks fails.